tests/unit/linalg/test_matrix_block.cpp

```diff
@@ -299,6 +299,60 @@ TEST_CASE("BlockMatrix", "[BlockMatrix]")
       REQUIRE(error < tol );
    }
 
+   Vector y1(A->Height());
+   Vector y2(Amono->Height());
+   SECTION("Check PartMult")
+   {
+      Array<int> rows{{10,39,509,289,1112,1321,927}};
```

Comment:
The matrices in this test seem larger than necessary, to verify the capabilities. Does this test take more than 1 second? If not, it's fine. If so, can the test matrices be made smaller?

Reply:
All the units for the `BlockMatrix` together take less than half a second